zerver/actions/user_groups.py

```diff
@@ -266,3 +268,20 @@ def check_delete_user_group(
     user_group = access_user_group_by_id(user_group_id, user_profile)
     user_group.delete()
     do_send_delete_user_group_event(user_profile.realm, user_group_id, user_profile.realm.id)
+
+
+def do_change_user_group_permission_setting(
+    user_group: UserGroup,
+    setting_name: str,
+    setting_value_group: UserGroup,
+    *,
+    acting_user: Optional[UserProfile],
+) -> None:
+    setattr(user_group, setting_name, setting_value_group)
+    user_group.save()
+
+    # RealmAuditLog changes are being done in a separate PR and will be
+    # added here once that is merged.
+    setting_id_name = setting_name + "_id"
+    event_data_dict: Dict[str, Union[str, int]] = {setting_id_name: setting_value_group.id}
+    do_send_user_group_update_event(user_group, event_data_dict)
```

Comment:
The event here is similar to the event send for name or description change of the user group. Do we want this to be something like we have for realm setting, where we have a `property` field set to setting name and `value` field set to the new setting value.

Reply:
Let's discuss that in the "api design" stream.

Reply:
Started discussion - https://chat.zulip.org/#narrow/stream/378-api-design/topic/event.20for.20updating.20group.20setting